ivy/functional/frontends/paddle/tensor/tensor.py

```diff
@@ -751,7 +751,11 @@ def nonzero(self):
 
     @with_supported_dtypes({"2.5.1 and below": ("float32", "float64")}, "paddle")
     def inner(self, y, name=None):
-        return paddle_frontend.inner(self, y, name)
+        return paddle_frontend.inner(self._ivy_array, y, name)
+
+    @with_unsupported_dtypes({"2.5.1 and below": ("float16", "bfloat16")}, "paddle")
```

Comment:
Looking at the [paddle docs](https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/acos_en.html#acos), seems like its better to use `with_supported_dtypes` decorator with float32, float64 and float16.

Reply:
@bipinKrishnan pls review my latest PR